pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/ChangeRequestConditional.java

```diff
@@ -0,0 +1,347 @@
+     */
+    public String getAuthorDisplayNameX() {
+        return authorDisplayNameX;
+    }
+
+    @DataBoundSetter
+    public void setAuthorDisplayNameX(String authorDisplayNameX) {
+        this.authorDisplayNameX = authorDisplayNameX;
+    }
+
+    /**
+     * CHANGE_AUTHOR_EMAIL ant style matching.
+     * @return authorEmail
+     * @see ContributorMetadataAction#getContributorEmail()
+     */
+    public String getAuthorEmail() {
+        return authorEmail;
+    }
+
+    @DataBoundSetter
+    public void setAuthorEmail(String authorEmail) {
+        this.authorEmail = authorEmail;
+    }
+
+    /**
+     * CHANGE_AUTHOR_EMAIL regular expression matching.
+     * @return authorEmailX
+     * @see ContributorMetadataAction#getContributorEmail()
+     */
+    public String getAuthorEmailX() {
+        return authorEmailX;
+    }
+
+    @DataBoundSetter
+    public void setAuthorEmailX(String authorEmailX) {
+        this.authorEmailX = authorEmailX;
+    }
+
+    public boolean matches(EnvVars vars) {
+        if (!CHANGE_ID.exists(vars)) {
+            return false; //Not a change
+        }
+        try {
+            check(id, Comparator.ANT, CHANGE_ID, vars);
+            check(target, Comparator.ANT, CHANGE_TARGET, vars);
+            check(branch, Comparator.ANT, CHANGE_BRANCH, vars);
+            check(fork, Comparator.ANT, CHANGE_FORK, vars);
+            check(urlX, Comparator.REG, CHANGE_URL, vars);
+            check(title, Comparator.ANT, CHANGE_TITLE, vars);
+            check(titleX, Comparator.REG, CHANGE_TITLE, vars);
+            check(author, Comparator.ANT, CHANGE_AUTHOR, vars);
+            check(authorX, Comparator.REG, CHANGE_AUTHOR, vars);
+            check(authorDisplayName, Comparator.ANT, CHANGE_AUTHOR_DISPLAY_NAME, vars);
+            check(authorDisplayNameX, Comparator.REG, CHANGE_AUTHOR_DISPLAY_NAME, vars);
+            check(authorEmail, Comparator.ANT, CHANGE_AUTHOR_EMAIL, vars);
+            check(authorEmailX, Comparator.REG, CHANGE_AUTHOR_EMAIL, vars);
+            return true;
+        } catch (AssertionError e) {
+            return false;
+        }
+    }
+
+    private void check(String pattern, Comparator comparator, Names varName, EnvVars vars) {
+        if (isNotEmpty(pattern)) {
+            if (!comparator.compare(pattern, varName.get(vars))) {
+                throw new AssertionError(varName.name() + "("+varName.get(vars)+")" + " does not match " + pattern);
+            }
+        }
+    }
+
+    @Extension
+    @Symbol("changeRequest")
+    public static class DescriptorImpl extends DeclarativeStageConditionalDescriptor<ChangeRequestConditional> {
+        @Override
+        public Expression transformToRuntimeAST(@CheckForNull ModelASTWhenContent original) {
+            return ASTParserUtils.transformWhenContentToRuntimeAST(original);
+        }
+    }
+
+    enum Comparator { //TODO this should be in a utility class for all when conditions to use
```

Comment:
So do so? =) No need to rewrite the other conditions to use it right away, but might as well start out with this in the place you want it to end up.